tribe-ext-ea-additional-options.php

```diff
@@ -0,0 +1,329 @@
+                        ></span>
+                </div>
+                <div class="tribe-refine tribe-active">
+                    <label for="tribe-ea-field-prefix"><?php esc_html_e('Event Title Prefix:', 'tribe-ext-ea-additional-options'); ?></label>
+                    <input id="tribe-ea-field-prefix" name="aggregator[prefix]" class="tribe-ea-field tribe-ea-size-large" type="text" value="<?php echo esc_attr($prefixValue); ?>" />
+                    <span
+                        class="tribe-bumpdown-trigger tribe-bumpdown-permanent tribe-bumpdown-nohover tribe-ea-help dashicons dashicons-editor-help"
+                        data-bumpdown="<?php echo esc_attr__('Add text before the title of each event.', 'tribe-ext-ea-additional-options'); ?>"
+                        data-width-rule="all-triggers"
+                        ></span>
+                </div>
+            </div>
+            <?php
+        }
+
+        /**
+         * Checks website url setting and filters link
+         * 
+         * @param string $link
+         * @param int $postId
+         * @return string
+         */
+        public function filter_event_link($link, $postId) {
+            $filterLinkOpt = tribe_get_option($this->opts_prefix . 'link_directly_to_website_url');
+            if ($filterLinkOpt === 'yes') {
+                $website_url = tribe_get_event_website_url($postId);
+                if (!empty($website_url)) {
+                    $link = $website_url;
+                }
+            }
+            return $link;
+        }
+
+        public function filter_service_data_field_map($fieldMap) {
+            $lineBreakOpt = tribe_get_option($this->opts_prefix . 'retain_line_breaks');
+            if ($lineBreakOpt == 'yes') {
+                if (isset($fieldMap['description'])) {
+                    unset($fieldMap['description']);
+                }
+                $fieldMap['unsafe_description'] = 'post_content';
+            }
+            return $fieldMap;
+        }
+
+        /**
+         * Filters event info before being saved or updated
+         * 
+         * @param array $event
+         * @return array
+         */
+        public function filter_imported_event($event, $record) {
+            $meta = $record->meta;
+            $lineBreakOpt = tribe_get_option($this->opts_prefix . 'retain_line_breaks');
+            if ($lineBreakOpt == 'yes') {
+                $event['post_content'] = str_replace(['\\n', '\n', "\n", "\\n"], '<br>', $event['post_content']);
+            }
+            if(!empty($meta['prefix'])){
+                $event['post_title'] = $meta['prefix'] . ' ' . $event['post_title'];
+            }
+            if(!empty($meta['timezone'])){
+                if(!empty($event['EventAllDay']) && $event['EventAllDay'] == 'yes'){
+                    $event['EventTimezone'] = $meta['timezone'];
+                }else{
+                    date_default_timezone_set( "UTC" );
```

Comment:
In this case you can create a timezone variable an use it instead as this might affect other areas or plugins.

```suggestion
                    $utc = new DateTimeZone( "UTC" );
```

Reply:
Good call. Thanks!